tools/revng-lift/JumpTargetManager.cpp

```diff
@@ -1658,7 +1575,27 @@ void JumpTargetManager::harvestWithAVI() {
   //
   // Update alias analysis
   //
-  aliasAnalysis();
+  {
+    FunctionAnalysisManager FAM;
+    ModuleAnalysisManager MAM;
+    MAM.registerPass([&] { return GeneratedCodeBasicInfoAnalysis(); });
+    MAM.registerPass([&] { return FunctionAnalysisManagerModuleProxy(FAM); });
+    FAM.registerPass([&] { return ModuleAnalysisManagerFunctionProxy(MAM); });
+
+    ModulePassManager MPM;
+    MPM.addPass(RequireAnalysisPass<GeneratedCodeBasicInfoAnalysis, Module>());
+    {
+      FunctionPassManager FPM;
+      FPM.addPass(CSVAliasAnalysisPass());
+      MPM.addPass(createModuleToFunctionPassAdaptor(std::move(FPM)));
+    }
+
+    PassBuilder PB;
+    PB.registerModuleAnalyses(MAM);
+    PB.registerFunctionAnalyses(FAM);
+
+    MPM.run(*M, MAM);
```

Comment:
This code is bulky, present in several parts of the code and might be worth factoring in a simpler/safer API.

Reply:
I understand, but just a small part of that code is used elsewhere, which is the registration of the PassBuilder. We could factor that one, but would likely make the pipeline-related code less readable, it's just two lines. Different things are registered by different objects (e.g., `GeneratedCodeBasicInfoAnalysis` here is registered by a `ModuleAnalysisManager`, elsewhere by a `FunctionAnalysisManager`). Same as regards the addition of passes to the pipeline. This particular pipeline is also different from the others considered that registers a pass (`CSVAAPass`) which is executed on every function of the module (via `createModuleToFunctionPassAdaptor`). Refactoring part of this code would shatter the pipeline and bring it less readable, considered that they're pretty different among each other (in the codebase).